content/en/blog/_posts/2022-07-27-crd-validation-rules-graduate-to-beta.md

```diff
@@ -0,0 +1,146 @@
+---
+layout: blog
+title: "Kubernetes 1.25: CustomResourceDefinition Validation Rules Graduate to Beta"
+date: 2022-07-27
+slug: tbd
+canonicalUrl: tbd
+---
+
+**Authors:** Joe Betz (Google), Kermit Alexander (Google)
+
+In Kubernetes 1.25, [Validation rules for CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules) (CRDs) have graduated to Beta!
+
+Validation rules make it possible to declare how custom resources are validated using the [Common Expression Language](https://github.com/google/cel-spec) (CEL). For example:
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+    ...
+    openAPIV3Schema:
+      type: object
+      properties:
+        spec:
+          type: object
+          x-kubernetes-validations:
+            - rule: "self.minReplicas <= self.replicas && self.replicas <= self.maxReplicas"
+              message: "replicas should be in the range minReplicas..maxReplicas."
+          properties:
+            replicas:
+              type: integer
+            ...
+```
+
+Validation rules support a wide range of use cases. To get a sense of some of the capabilities, let's look at a few examples:
+
+
+| Validation Rule | Purpose |
+| - | - |
+| `self.minReplicas <= self.replicas` | Validate an integer field is less than or equal to another integer field |
+| `'Available' in self.stateCounts` | Validate an entry with the 'Available' key exists in a map |
+| `self.set1.all(e, !(e in self.set2))` | Validate that the elements of two sets are disjoint |
+| `self == oldSelf` | Validate that a required field is immutable once it is set |
+| `self.created + self.ttl < self.expired` | Validate that 'expired' date is after a 'create' date plus a 'ttl' duration |
+
+
+Validation rules are expressive and flexible. See the [Validation Rules documentation](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules) to learn more about what validation rules are capable of.
```

Comment:
```suggestion
Validation rules are expressive and flexible. See the [Validation Rules documentation](/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules) to learn more about what validation rules are capable of.
```



Ideally, link to the section about validation rules in the CRD concept docs (if there isn't a section yet, please mention the PR that'll add it so that a reviewer can cross-check).